Content.Server/Station/Systems/StationSystem.cs

```diff
@@ -94,11 +94,33 @@ private void OnStationStartup(EntityUid uid, StationDataComponent component, Com
 
     private void OnStationDeleted(EntityUid uid, StationDataComponent component, ComponentShutdown args)
     {
-        _stations.Remove(uid);
+        if (_stations.Contains(uid) && // Was not deleted via DeleteStation() 
+            _gameTicker.RunLevel == GameRunLevel.InRound) // And not due to a round restart
+        {
+            throw new InvalidOperationException($"Station entity {ToPrettyString(uid)} is getting deleted mid-round.");
+        }
```

Comment:
Would be nice to remove this at some stage if we want to delete maps at runtime.

Reply:
There is a `DeleteStation()` function, which removes entities from the _stations list before deleting the entity, though AFAIK its unused atm. Hence the `_stations.Contains(uid)` check. So while its somewhat hacky, it should already be possible